asset-transfer-private-data/chaincode-go/private_asset_queries.go

```diff
@@ -0,0 +1,193 @@
+		}
+		if buyerIdentity == nil {
+			return "", fmt.Errorf("%s does not exist", assetID)
+		}
+
+	return string(buyerIdentity), nil
+
+}
+
+// ===========================================================================================
+// GetAssetByRange performs a range query based on the start and end keys provided. Range
+// queries can be used to read data from private data collections, but can not be used in
+// a transaction that also writes to private data.
+
+// ===========================================================================================
+func (s *SmartContract) GetAssetByRange(ctx contractapi.TransactionContextInterface, startKey string, endKey string) ([]Asset, error) {
+
+	resultsIterator, err := ctx.GetStub().GetPrivateDataByRange("assetCollection", startKey, endKey)
+	if err != nil {
+			return nil, err
+	}
+	defer resultsIterator.Close()
+
+	results := []Asset{}
+
+	for resultsIterator.HasNext() {
+		response, err := resultsIterator.Next()
+		if err != nil {
+			return nil, err
+		}
+
+		newAsset := new(Asset)
+
+		err = json.Unmarshal(response.Value, newAsset)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+		}
+
+		results = append(results, *newAsset)
+	}
+
+	return results, nil
+
+}
+
+
+// =======Rich queries =========================================================================
+// Two examples of rich queries are provided below (parameterized query and ad hoc query).
+// Rich queries pass a query string to the state database.
+// Rich queries are only supported by state database implementations
+//  that support rich query (e.g. CouchDB).
+// The query string is in the syntax of the underlying state database.
+// With rich queries there is no guarantee that the result set hasn't changed between
+//  endorsement time and commit time, aka 'phantom reads'.
+// Therefore, rich queries should not be used in update transactions, unless the
+// application handles the possibility of result set changes between endorsement and commit time.
+// Rich queries can be used for point-in-time queries against a peer.
+// ============================================================================================
+
+// ===== Example: Parameterized rich query =================================================
+// QueryAssetByOwner queries for assets based on a passed in owner.
+// This is an example of a parameterized query where the query logic is baked into the chaincode,
+// and accepting a single query parameter (owner).
+// Only available on state databases that support rich query (e.g. CouchDB)
+// =========================================================================================
+func (s *SmartContract) QueryAssetByOwner(ctx contractapi.TransactionContextInterface, owner string) ([]Asset, error) {
+
+	ownerString  := strings.ToLower(owner)
+
+	queryString := fmt.Sprintf("{\"selector\":{\"docType\":\"asset\",\"owner\":\"%s\"}}", ownerString)
```

Comment:
"docType" > "type" , since the asset prop got rename

Reply:
done.